lib/archive/unpack.go

```diff
@@ -91,124 +94,34 @@ func Unpack(op trace.Operation, tarStream io.Reader, filter *FilterSpec, root st
 		absPath := filepath.Join(root, rebased)
 		op.Debugf("attempting to write to target (%s)", absPath)
 
+		op.Debugf("writing to: %s", absPath)
+
 		switch header.Typeflag {
 		case tar.TypeDir:
 			err = os.MkdirAll(absPath, header.FileInfo().Mode())
 			if err != nil {
 				return err
 			}
-			continue
 		case tar.TypeSymlink:
-
 			err := os.Symlink(header.Linkname, absPath)
 			if err != nil {
 				op.Errorf("Failed to create symlink %s->%s: %s", absPath, header.Linkname, err)
 				return err
 			}
-
 		case tar.TypeReg:
 			f, err := os.OpenFile(absPath, fileWriteFlags, header.FileInfo().Mode())
 			if err != nil {
 				return err
 			}
-
 			_, err = io.Copy(f, tr)
 			// TODO: add ctx.Done cancellation
 			f.Close()
 			if err != nil {
 				return err
 			}
-
 		default:
 			// TODO: add support for special file types - otherwise we will do absurd things such as read infinitely from /dev/random
 		}
 	}
 	return nil
 }
-
-func processFirstEntry(op trace.Operation, tarStream *tar.Reader, filter *FilterSpec, root string) error {
```

Comment:
yeah with ContainerStatPath running well there is no longer a need for this hack :)